backend/src/utils/cache.py

```diff
@@ -0,0 +1,46 @@
+# I SPENT 2 HOURS FIGURING THIS OUT FML
+from typing import Any
+
+import jsoncache
+import valkey
+from valkey.exceptions import (
+    ConnectionError as ValkeyConnectionError,  # I HATE THIS STUPID BS WHY ARE THERE MULTIPLE CONNECTION ERRORS ISWTG THIS IS THE STUPIDEST BUG EVER,
+)
+
+conn = valkey.Valkey()
+
+
+async def set_cache(data: dict[str, Any], title: str, write_json: bool) -> None:
+
+    try:
+        await conn.ping()
```

Comment:
```suggestion
        await conn.set(title, json.dumps(data))
```
you can't store just a dict in a valkey, so serialize it to string and convert it back to dictionary in get_cache.